src/humanloop/otel/__init__.py

```diff
@@ -0,0 +1,80 @@
+from opentelemetry import baggage
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.trace import Tracer
+
+from humanloop.otel.constants import HL_TRACE_METADATA_KEY
+from humanloop.otel.helpers import module_is_installed
+
+
+_TRACER = None
+
+_BAGGAGE_CONTEXT = [{}]
+
+
+def set_tracer(tracer: Tracer):
+    global _TRACER
+    _TRACER = tracer
```

Comment:
Why is this needed now that its in the client?

Reply:
added more explanations in that module, will follow up with an elaborate description on PR

Reply:
This is the Baggage point that I've mentioned in the PR description

Reply:
Still not understanding why this state can't be managed on the client, instead of as a global

Reply:
+1

i think this should be client-specific.

Reply:
ack, fixed

Reply:
@harry-humanloop can you check the fix?

Reply:
@harry-humanloop 
The "raw" decorators expect the tracer as a parameter:

https://github.com/humanloop/humanloop-python/pull/24/files#diff-898818d52efe202dd7599b3137839bf8fbbad7a8dd5c05636419d5e1ed2a2f67R25

On the decorators exposed to the developer via the client we inject the Tracer instance available to the client instance:
https://github.com/humanloop/humanloop-python/pull/24/files#diff-bda3268f707012d2525cd66f2815381a1c67080c946c894dbd5bc7ac5117acdfR283